LDAPCollector.class.inc.php

```diff
@@ -154,15 +154,40 @@ public function Search($sDN, $sFilter, $aAttributes = array('*'))
         if ($this->Connect())
         {
             Utils::Log(LOG_DEBUG, "ldap_search('$sDN', '$sFilter', ['".implode("', '", $aAttributes)."'])...");
+
+	$cookie = '';
+	$pageSize = 500;
+	$firstPass = true;
+	do {
+	    ldap_control_paged_result($this->rConnection, $pageSize, true, $cookie);
+
             $rSearch = @ldap_search($this->rConnection, $sDN, $sFilter, $aAttributes);
             if ($rSearch === false)
             {
                 Utils::Log(LOG_ERR, "ldap_search('$sDN', '$sFilter') FAILED (".ldap_error($this->rConnection).").");
                 return false;
             }
             Utils::Log(LOG_DEBUG, "ldap_search() Ok.");
+
+	    $bList = ldap_get_entries($this->rConnection, $rSearch);
+	    if($firstPass)
+	    {
+		$aList = $bList;
+	    }
+	    else
+	    {
+		$aList = array_merge($aList, $bList);
+	    }
+
+	    ldap_control_paged_result_response($this->rConnection, $rSearch, $cookie);
```

Comment:
Same remark, this is deprecated
https://www.php.net/manual/en/function.ldap-control-paged-result-response